File: src/test/java/com/orange/callhistory/service/CallServiceUTest.java
package com.orange.callhistory.service;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

import com.orange.callhistory.dao.CallDao;
import com.orange.callhistory.dao.CallRepository;
import org.junit.Test;

public class CallServiceUTest {

    private final CallRepository callRepository = mock(CallRepository.class);

    private final CallService callService = new CallService(callRepository);

    private final Call call = new Call("C1", null, null, null);

    @Test
    public void should_save_a_call() {
        callService.save(call);
        verify(callRepository).save(any(CallDao.class));
    }


    @Test
    public void should_find_a_call() {
        callService.findCall("C1");
        verify(callRepository).findById("C1");
    }
}

File: src/test/java/com/orange/callhistory/service/CallServiceUTest.kt
package com.orange.callhistory.service

import com.orange.callhistory.dao.CallDao
import com.orange.callhistory.dao.CallRepository
import org.junit.Test
import org.mockito.ArgumentMatchers.any
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

class CallServiceUTest {

    private val callRepository = mock<CallRepository>(CallRepository::class.java)

    private val callService = CallService(callRepository)

    private val call = Call("C1", "", "", 0)

    @Test
    fun should_save_a_call() {
        callService.save(call)
        verify(callRepository).save(any(CallDao::class.java!!))
    }


    @Test
    fun should_find_a_call() {
        callService.findCall("C1")
        verify(callRepository).findById("C1")
    }
}